SimG4Core/MagneticField/src/Field.cc

```diff
@@ -1,74 +1,48 @@
 #include "SimG4Core/MagneticField/interface/Field.h"
 #include "MagneticField/Engine/interface/MagneticField.h"
 
-//#include "Geometry/Vector/interface/GlobalPoint.h"
 #include "DataFormats/GeometryVector/interface/GlobalPoint.h"
-
 #include "G4Mag_UsualEqRhs.hh"
 
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-#include "SimG4Core/Notification/interface/SimG4Exception.h"
-#include "FWCore/Utilities/interface/isFinite.h"
-
 using namespace sim;
 
 G4Mag_UsualEqRhs * Field::fieldEquation() { return theFieldEquation; }
 
 Field::Field(const MagneticField * f, double d) 
-    : G4MagneticField(), theCMSMagneticField(f),theDelta(d)
+  : G4MagneticField(), theCMSMagneticField(f), theDelta(d)
 {
+  for(int i=0; i<3; ++i) {
+    oldx[i] = 1.0e12;
+    oldb[i] = 0.0;
+  }
 }
 
 Field::~Field() {}
 
-void Field::GetFieldValue(const double xyz[3],double bfield[3]) const 
+void Field::GetFieldValue(const G4double xyz[4], double* bfield) const 
```

Comment:
@davidlange6 , I simply implement exact interface of G4MagneticField. Few years ago G4 interface was changed. In this CMS class the old version is implemented. However, both are equivalent and this change is not important. The main modifications concern removal of unnecessary operations which were were performed several times at each simulation step: call to tls and check on nan.
More serious modification for magnetic field is in  a progress so , please, merge this PR, so I can propose the next one.